lib/ai/model/agent.go

```diff
@@ -47,16 +48,56 @@ const (
 )
 
 // NewAgent creates a new agent. The Assist agent which defines the model responsible for the Assist feature.
-func NewAgent(assistClient assist.AssistEmbeddingServiceClient, username string) *Agent {
-	return &Agent{
-		tools: []Tool{
-			&commandExecutionTool{},
+func NewAgent(username string, config ToolsConfig) (*Agent, error) {
+	err := config.CheckAndSetDefaults()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	tools := []Tool{&commandExecutionTool{}}
+
+	if !config.DisableEmbeddingsTool {
```

Comment:
This looks a bit hacky. Now if you pass `DisableEmbeddingsTool` you also must pass `username` which won't be used.